sqldelight-compiler/dialect/src/main/kotlin/app/cash/sqldelight/dialect/api/SqlDelightDialect.kt

```diff
@@ -19,9 +19,9 @@ interface SqlDelightDialect {
    */
   val asyncRuntimeTypes: RuntimeTypes
```

Comment:
I don't think we need this in its current form anymore since the drivers can now be used interchangeably with both the blocking and async codegen, _but_ if you're generating code for postgres or mysql the codegen will generate everything with JDBC-specific classes. We'd need some other way of controlling the types that are used in the codegen to use R2DBC classes instead for example.

Reply:
I think that's okay - what we should do is allow the JDBC driver to be extendable so that you could have R2DBC extend from that. At least that is how I am imagining this would work now that its a shared runtime - there is still a single driver/cursor/preparedstatement type, but you can subclass those types to change how it would actually execute queries

Reply:
To me it seems like a questionable use of inheritance since there's no real relationship between the JDBC and R2DBC (or any other possible JVM-based driver like jasync). Right now the `JdbcDriver` and its related classes are set up assuming you'll have a JDBC `Connection`, `PreparedStatement`, and `ResultSet` instances to use which don't exist in R2DBC. If we tried modifying the `JdbcDriver` to account for different types of those classes, then I don't see how that's any different than implementing the base interfaces like we do now.

Maybe a new type of dialect module that can override the generated types would be a better way to approach this?

Reply:
yep yep yep good points, dialect might be a good approach to it but lets leave it as it is now and we can revisit later